### ISSUE TITLE
vscode solution builder catalog explorer snippets are not parsed by ibm blueprints

### ISSUE THREAD

**Issue by powellquiring** [IBM/solution-builder-vscode-extension#6]
**Describe the bug**
1. Paste a snippet into a blueprint t.yaml file:
![image](https://user-images.githubusercontent.com/6932057/218197505-e072e7c6-deca-4c22-a6c0-3738a1d29777.png)
1. add value: to create a valid blueprint
1. Parse the file with blueprints:

Example input:
```
name: "Blueprint Basic Example"
# yaml-language-server: $schema=https://raw.githubusercontent.com/Cloud-Schematics/vscode-blueprint-schema/master/blueprint_schema.json
type: "blueprint"
schema_version: "1.0.0"
description: "Simple blueprint to demonstrate module linking"
inputs:
  - name: x
settings:
  - name: TF_VERSION
    value: 1.0
modules:
 - name: ibm-watson-studio
   module_type: terraform
   source:
     git:
       source_type: github
       git_repo_url: "https://github.com/terraform-ibm-modules/terraform-ibm-toolkit-watson-studio"
   settings:
     - name: TF_VERSION
       value: "1.2"
   inputs:
     - name: resource_group_name
       value: $blueprint.x
       description: Resource group where the cluster has been provisioned.
       type: string
     - name: resource_location
       value: $blueprint.x
       description: Geographic location of the resource (e.g. us-south, us-east)
       type: string
     - name: tags
       value: $blueprint.x
       description: Tags that should be applied to the service
       type: list(string)
       default: []
     - name: name_prefix
       value: $blueprint.x
       description: The prefix name for the service. If not provided it will default to
         the resource group name
       type: string
     - name: plan
       value: $blueprint.x
       description: The type of plan the service instance should run under (lite,
         professional-v1)
       type: string
       default: professional-v1
     - name: name
       value: $blueprint.x
       description: The name of the Watson Studio instance that will be provisioned
       type: string
     - name: label
       value: $blueprint.x
       description: The label of the resource used to build the name along with the
         name_prefix.
       type: string
       default: studio
   outputs:
     - name: id
       description: The id of the provisioned redis instance
     - name: name
       description: The name of the provisioned redis instance
     - name: crn
       description: The id of the provisioned redis instance
     - name: location
       description: The location of the provisioned redis instance
     - name: service
       description: The name of the key provisioned for the redis instance
     - name: label
       description: The label for the redis instance
```
Parse errors:
```
ibmcloud schematics blueprint create -name t00 -resource-group feedback -bp-git-url https://github.com/powellquiring/blueprint-basic-example -bp-git-file fun_blueprint.yaml
blueprint-basic-example $ ibmcloud schematics blueprint create -name t00 -resource-group feedback -bp-git-url https://github.com/powellquiring/blueprint-basic-example -bp-git-file
t.yaml
FAILED
Could not create blueprint. Please verify that your request is correct. If the problem persists, contact IBM Cloud support.

messageid:	M1156
requestid:	38d413e4-2c31-4d38-9673-a180547c138f
statuscode:	400
timestamp:	2023-02-10T21:09:02.734196162Z
errors:	[yaml: unmarshal errors:
  line 19: field source_type not found in type models.UBGit
  line 37: cannot unmarshal !!seq into string
  line 61: field description not found in type models.UBComponentOutput
  line 63: field description not found in type models.UBComponentOutput
  line 65: field description not found in type models.UBComponentOutput
  line 67: field description not found in type models.UBComponentOutput
  line 69: field description not found in type models.UBComponentOutput
  line 71: field description not found in type models.UBComponentOutput]
message:	Invalid blueprint definitions. Error - Unmarshal error
```

Summary
  line 19: field source_type not found in type models.UBGit
```
 - name: ibm-watson-studio
   module_type: terraform
   source:
     git:
       source_type: github
       git_repo_url: "https://github.com/terraform-ibm-modules/terraform-ibm-toolkit-watson-studio"
```
  line 37: cannot unmarshal !!seq into string
default: []
```
     - name: tags
       value: $blueprint.x
       description: Tags that should be applied to the service
       type: list(string)
       default: []
```

  line 61: field description not found in type models.UBComponentOutput

```
   outputs:
     - name: id
       description: The id of the provisioned redis instance
     - name: name
```

**Steps to reproduce**
Steps above to create the blueprint yaml, t.yaml.  Then use it:
```
$ ibmcloud schematics blueprint create -name t00 -resource-group feedback -bp-git-url https://github.com/powellquiring/blueprint-basic-example -bp-git-file t.yaml --inputs x=x
```

**Comment by nfstein** [comment]
I've opened a defect with the blueprint team for the unmarshal error you were encountering for the default: [] line. Looking into the errors for source_type and  output description lines, we've discovered that the blueprint team is on a different version of the schema than we are and we are awaiting confirmation that their version is the preferred one going forward.

**Comment by nfstein** [comment]
patch included in v0.3.4 release
the default: [] issue has been raised with the blueprint team and is not included in the patch because it is valid blueprint as included
